release/src/main/groovy/run_release_candidate_python_validation.sh

```diff
@@ -0,0 +1,272 @@
+
+print_separator "Running wordcount example with DirectRunner"
+python -m apache_beam.examples.wordcount --output wordcount_direct.txt
+file="wordcount_direct.txt-00000-of-00001"
+if [ -f "$file" ]
+then
+	echo "$file found."
+else
+	echo "ERROR: $file not found."
+	complete "failed when running wordcount example with DirectRunner."
+	exit 1
+fi
+echo "SUCCEED: wordcount successfully run on DirectRunner."
+
+
+#
+# 5. Run wordcount with DataflowRunner
+#
+
+print_separator "Running wordcount example with DataflowRunner "
+python -m apache_beam.examples.wordcount \
+    --output gs://$BUCKET_NAME/$WORDCOUNT_OUTPUT \
+    --staging_location gs://$BUCKET_NAME$TEMP_DIR \
+    --temp_location gs://$BUCKET_NAME$TEMP_DIR \
+    --runner DataflowRunner \
+    --job_name wordcount \
+    --project $PROJECT_ID \
+    --num_workers $NUM_WORKERS \
+    --sdk_location dist/apache-beam-$VERSION.tar.gz
+
+# verify results.
+gcs_pull_result=$(gsutil ls gs://$BUCKET_NAME)
+for index in {0..3}
+do
+    if [[ $gcs_pull_result != *"gs://$BUCKET_NAME/$WORDCOUNT_OUTPUT-0000$index-of-00004"* ]]
+    then
+        echo "ERROR: The wordcount example failed on DataflowRunner".
+        complete "failed when running wordcount example with DataflowRunner."
+        exit 1
+    fi
+done
+# clean output files from GCS
+gsutil rm gs://$BUCKET_NAME/$WORDCOUNT_OUTPUT-*
+echo "SUCCEED: wordcount successfully run on DataflowRunner."
+
+
+#
+# 6. Run Streaming wordcount with DirectRunner
+#
+
+create_pubsub
+print_separator "Running Streaming wordcount example with DirectRunner"
+python -m apache_beam.examples.streaming_wordcount \
+--input_topic projects/$STREAMING_PROJECT_ID/topics/$PUBSUB_TOPIC1 \
+--output_topic projects/$STREAMING_PROJECT_ID/topics/$PUBSUB_TOPIC2 \
+--streaming &
+pid=$!
+sleep 15
+
+# verify result
+run_pubsub_publish
+pull_result=$(run_pubsub_pull)
+should_see="Python: "
+if [[ $pull_result = *"$should_see"* ]]
+then
+    echo "SUCCEED: The streaming wordcount example running successfully on DirectRunner."
+else
+    echo "ERROR: The streaming wordcount example failed on DirectRunner."
+    cleanup_pubsub
+    kill -9 $pid
+    complete "failed when running streaming wordcount example with DirectRunner."
+    exit 1
+fi
+kill -9 $pid
+sleep 10
+
+
+#
+# 7. Run Streaming Wordcount with DataflowRunner
+#
+
+print_separator "Running Streaming wordcount example with DirectRunner "
+python -m apache_beam.examples.streaming_wordcount \
+    --streaming \
+    --job_name pyflow-wordstream-candidate \
+    --project $STREAMING_PROJECT_ID \
+    --runner DataflowRunner \
+    --input_topic projects/$STREAMING_PROJECT_ID/topics/$PUBSUB_TOPIC1 \
+    --output_topic projects/$STREAMING_PROJECT_ID/topics/$PUBSUB_TOPIC2 \
+    --staging_location gs://$STREAMING_BUCKET_NAME$STREAMING_TEMP_DIR \
+    --temp_location gs://$STREAMING_BUCKET_NAME$STREAMING_TEMP_DIR \
+    --num_workers $NUM_WORKERS \
+    --sdk_location dist/apache-beam-$VERSION.tar.gz &
+
+pid=$!
+sleep 60
```

Comment:
60 seconds may or may not be enough. (Should be enough though.) Safer way would be to see whether a subscription to the input topic is created or not.

Reply:
Another option is to create an input subscription and use that subscription in the pipeline. Short of that, 60 seconds should be enough. We can increase it later if we see an issue.